### PR TITLE
mgr/dashboard: Add E2E tests for navigation, language and login

### PR DESCRIPTION
**Pull request by tspmelo**
Fixes: https://tracker.ceph.com/issues/45376

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/cypress/integration/ui/navigation.po.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  getVerticalMenu() {
+    return cy.get('ul.cd-navbar-primary');
+  }
+
+  getMenuToggler() {
+    return cy.get('cd-navigation > div.cd-navbar-top button.btn.btn-link');
+  }
+
+  checkNavigations(navs: any) {
+    // This requests are mocked for a faster response and preventing blocking the test
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/frontend/cypress/integration/ui/navigation.po.ts:55]: Are all requests mocked or only the nfs-ganesha and RGW status? As it looks like the latter, could you may be change the comment to
```suggestion
    // The nfs-ganesha and RGW status requests are mocked to ensure that this method runs in time
```

Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/cypress/integration/ui/navigation.po.ts]: Done

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by callithea** [comment]
jenkins test dashboard backend

**Pull request merged**